backend/apps/notifications/views.py

```
@@ -192,3 +215,128 @@ def post(self, request):
+            delivery.opened_at = timezone.now()
+            delivery.save(update_fields=["status", "opened_at", "updated_at"])
+        except NotificationDelivery.DoesNotExist:
+            pass
+
+        pixel = b'GIF89a\x01\x00\x01\x00\x80\x00\x00\xff\xff\xff\x00\x00\x00!\xf9\x04\x01\x00\x00\x00\x00,\x00\x00\x00\x00\x01\x00\x01\x00\x00\x02\x02D\x01\x00;'
+        return HttpResponse(pixel, content_type="image/gif")
+
+
+class TrackClickView(APIView):
+    """GET /api/notifications/track/click/<delivery_id>/?target=<url>"""
+    permission_classes = [permissions.AllowAny]
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)